src/app/components/challenge/challenge.component.ts

```diff
@@ -93,6 +116,147 @@ export class ChallengeComponent implements OnInit {
+        BODY
+        ).subscribe(
+        data => {
+          if (isPublished) {
+            this.publishChallenge.state = 'Published';
+            this.publishChallenge.icon = 'fa fa-eye green-text';
+          } else {
+            this.publishChallenge.state = 'Not Published';
+            this.publishChallenge.icon = 'fa fa-eye-slash red-text';
+          }
+          SELF.globalService.showToast('success', 'The challenge was successfully made ' + toggleChallengePublishState, 5);
+        },
+        err => {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () => console.log('PUBLISH-CHALLENGE-UPDATE-FINISHED')
+      );
+    };
+
+    const PARAMS = {
+      title: 'Make this challenge ' + toggleChallengePublishState + '?',
+      content: '',
+      confirm: 'Yes, I\'m sure',
+      deny: 'No',
+      confirmCallback: apiCall
+    };
+    SELF.globalService.showConfirm(PARAMS);
+  }
+
+  /**
+   * Edit challenge title function
+   */
+  editChallengeTitle() {
+    const SELF = this;
+
+    const apiCall = (params) => {
+      const BODY = JSON.stringify(params);
+      SELF.apiService.patchUrl(
+        SELF.endpointsService.editChallengeDetailsURL(SELF.challenge.creator.id, SELF.challenge.id),
+        BODY
+        ).subscribe(
+        data => {
+          SELF.challenge.title = data.title;
+          SELF.globalService.showToast('success', 'The challenge title is  successfully updated!', 5);
+        },
+        err => {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () => console.log('PUBLISH-CHALLENGE-UPDATE-FINISHED')
+      );
+    };
+
+    const PARAMS = {
+      title: 'Edit Challenge Title',
+      content: '',
+      confirm: 'Submit',
+      deny: 'Cancel',
+      form: [
+        {
+          name: 'editChallengeTitle',
+          isRequired: true,
+          label: 'title',
+          placeholder: 'Challenge Title',
+          type: 'text',
+          value: this.challenge.title
+        },
+      ],
+      confirmCallback: apiCall
+    };
+    SELF.globalService.showModal(PARAMS);
+  }
+
+  /**
+   * Delete challenge
+   */
+  deleteChallenge() {
+    const SELF = this;
+    const redirectTo = '/dashboard';
+
+    const apiCall = () => {
+      const BODY = JSON.stringify({});
+      SELF.apiService.postUrl(
+        SELF.endpointsService.deleteChallengeURL(SELF.challenge.id),
+        BODY
+        ).subscribe(
+        data => {
+          SELF.router.navigate([redirectTo]);
+          SELF.globalService.showToast('success', 'The Challenge is successfully deleted!', 5);
+        },
+        err => {
+          SELF.globalService.handleApiError(err, true);
+          SELF.globalService.showToast('error', err);
+        },
+        () => console.log('PUBLISH-CHALLENGE-UPDATE-FINISHED')
```

Comment:
Minor: Here, it should be PUBLISH-CHALLENGE-DELETE-FINISHED
Moreover, I guess all these console logs would need to be removed when we push it to production.

Reply:
@lunayach so should I remove all these?

Reply:
I guess you could keep them for now while we/you test the application before deployment. But please don't forget to remove these afterwards.

Reply:
@lunayach done